hazelcast/src/test/java/com/hazelcast/replicatedmap/impl/record/LazyIteratorTest.java

```diff
@@ -302,34 +317,35 @@ public void test_lazy_keyset_with_tombs_with_has_next() {
             count++;
             values.add(iterator.next());
         }
-        assertEquals(50, count);
-        assertEquals(50, values.size());
+        assertEqualsStringFormat("Expected %d items in the LazySet.iterator(), but was %d", 50, count);
+        assertEqualsStringFormat("Expected %d unique items in the LazySet.iterator(), but was %d", 50, values.size());
+        assertFalse("Expected no more items in LazySet.iterator()", iterator.hasNext());
     }
 
     @Test
-    public void test_lazy_keyset_with_tombs_with_next() {
-        KeySetIteratorFactory<String, Integer> factory = new KeySetIteratorFactory<String, Integer>(REPLICATED_RECORD_STORE);
+    public void testLazySet_withKeySetIterator_withTombs_next() {
+        KeySetIteratorFactory<String, Integer> factory = new KeySetIteratorFactory<String, Integer>(replicatedRecordStore);
         LazySet<String, Integer, String> collection = new LazySet<String, Integer, String>(factory, TEST_DATA_TOMBS);
         Iterator<String> iterator = collection.iterator();
 
         Set<String> values = new HashSet<String>();
         for (int i = 0; i < 50; i++) {
             values.add(iterator.next());
         }
-        assertEquals(50, values.size());
-
+        assertEqualsStringFormat("Expected %d unique items in the LazySet.iterator(), but was %d", 50, values.size());
+        assertFalse("Expected no more items in LazySet.iterator()", iterator.hasNext());
```

Comment:
This additional `iterator.hasNext()` call sets an internal field, which then prevents the `NoSuchElementException` from being thrown in the following `iterator.next()` call.